packages/flutter/lib/src/material/material_app.dart

```diff
@@ -73,6 +73,8 @@ class _MaterialAppState extends State<MaterialApp> implements BindingObserver {
   GlobalObjectKey _navigator;
 
   Size _size;
+  double _devicePixelRatio;
```

Comment:
I'm a little skeptical about the value of duplicating these values here (and having them in didChangeMetrics). We should probably just pass them through to the media query straight from the window. @abarth may have a more informed opinion about that, though.